[Safe CPP] Address Warnings in RenderFrame and HTMLFrameElement

stwrt opened the pull request:
#### 4af4465d5f452f743170d238635d5ffc429e4383
<pre>
[Safe CPP] Address Warnings in RenderFrame and HTMLFrameElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=293931">https://bugs.webkit.org/show_bug.cgi?id=293931</a>
<a href="https://rdar.apple.com/problem/152464519">rdar://problem/152464519</a>

Reviewed by Chris Dumez.

Address safe cpp warnings in RenderFrame and HTMLFrameElement.

* Source/WebCore/html/HTMLFrameElement.cpp:
(WebCore::HTMLFrameElement::attributeChanged):
* Source/WebCore/html/HTMLFrameElement.h:
* Source/WebCore/rendering/RenderFrame.cpp:
(WebCore::RenderFrame::frameElement const):
(WebCore::RenderFrame::edgeInfo const):
* Source/WebCore/rendering/RenderFrame.h:
(WebCore::HTMLFrameElement::renderer const):

Canonical link: <a href="https://commits.webkit.org/295787@main">https://commits.webkit.org/295787@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d1e96c33e8429e4cea0e357d0ffda72163c15da6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106075 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25824 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16218 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111272 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56671 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108114 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26480 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34327 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80576 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109079 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20923 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95727 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60898 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20503 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13825 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56109 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90291 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13860 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114127 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33213 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24508 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89657 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91955 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89346 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22788 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34215 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12023 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28786 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33138 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38550 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32884 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36234 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34482 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/HTMLFrameElement.h:
```diff
@@ -38,7 +39,7 @@ class HTMLFrameElement final : public HTMLFrameElementBase {
     bool noResize() const;
 
-    RenderFrame* renderer() const;
+    CheckedPtr<RenderFrame> renderer() const;
```
This is against our style. Our getters do not return smart pointers. Sometimes, we do have a *second* getter which is named `protectedFoo()` or `checkedFoo()` which return a Ref or a CheckedRef. I'm not sure this is helpful here since this is nullable and the caller will likely have to store the result in a local variable (which can be a CheckedPtr) in order to null check it.

stwrt replied:
Addressed the requested style changes.

cdumez commented on Source/WebCore/rendering/RenderFrame.h:
```diff
@@ -37,7 +39,7 @@ class RenderFrame final : public RenderFrameBase {
     virtual ~RenderFrame();
 
-    HTMLFrameElement& frameElement() const;
+    CheckedRef<HTMLFrameElement> frameElement() const;
```
against our style.

cdumez commented on Source/WebCore/rendering/RenderFrame.h:
```diff
@@ -48,7 +50,7 @@ class RenderFrame final : public RenderFrameBase {
 };
 
-inline RenderFrame* HTMLFrameElement::renderer() const
+inline CheckedPtr<RenderFrame> HTMLFrameElement::renderer() const
```
against our style.

nt1m commented on Source/WebCore/rendering/RenderFrame.cpp:
```diff
@@ -50,7 +51,8 @@ HTMLFrameElement& RenderFrame::frameElement() const
 FrameEdgeInfo RenderFrame::edgeInfo() const
 {
-    return FrameEdgeInfo(frameElement().noResize(), frameElement().hasFrameBorder());
+    CheckedRef frameElement = this->frameElement();
```
Is there a reason to use CheckedRef over just `Ref` ? elements can be ref counted

stwrt replied:
Switched to Ref

nt1m commented on Source/WebCore/html/HTMLFrameElement.cpp:
```diff
@@ -56,7 +58,8 @@ bool HTMLFrameElement::rendererIsNeeded(const RenderStyle& style)
 {
-    return createRenderer<RenderFrame>(*this, WTFMove(style));
+    CheckedRef renderStyle = style;
+    return createRenderer<RenderFrame>(*this, WTFMove(renderStyle.get()));
```
The guideline for this case is to protect at the call site, not use protectors.

cdumez replied:
The pre-existing code was totally safe here, this was a false positive. Ryosuke is aware and is working on fixing it. In the mean time, we should silence the warning, not try and protect with a local variable.

stwrt replied:
@cdumez Thanks for the update on the false positive. Reverted.

stwrt replied:
Added a suppression in the mean time.

nt1m commented on Source/WebCore/rendering/RenderFrame.cpp:
```diff
@@ -29,6 +29,8 @@
 #include "RenderBoxModelObjectInlines.h"
 #include "RenderFrameSet.h"
 #include "RenderWidgetInlines.h"
+#include <wtf/CheckedRef.h>
```
The CheckedRef include is no longer needed

stwrt committed:
[Safe CPP] Address Warnings in RenderFrame and HTMLFrameElement
https://bugs.webkit.org/show_bug.cgi?id=293931
rdar://problem/152464519

Reviewed by Chris Dumez.

Address safe cpp warnings in RenderFrame and HTMLFrameElement.

* Source/WebCore/html/HTMLFrameElement.cpp:
(WebCore::HTMLFrameElement::attributeChanged):
* Source/WebCore/html/HTMLFrameElement.h:
* Source/WebCore/rendering/RenderFrame.cpp:
(WebCore::RenderFrame::frameElement const):
(WebCore::RenderFrame::edgeInfo const):
* Source/WebCore/rendering/RenderFrame.h:
(WebCore::HTMLFrameElement::renderer const):

Canonical link: https://commits.webkit.org/295787@main
